supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/caching/CacheActor.scala

```diff
@@ -0,0 +1,62 @@
+package cromwell.backend.impl.htcondor.caching
+
+import akka.actor.{Actor, ActorLogging}
+import cromwell.backend.BackendJobExecutionActor.SucceededResponse
+import cromwell.backend.impl.htcondor.caching.CacheActor._
+import cromwell.backend.impl.htcondor.caching.exception.{CachedResultAlreadyExistException, CachedResultNotFoundException}
+import cromwell.backend.impl.htcondor.caching.model.CachedExecutionResult
+
+object CacheActor {
+
+  trait CacheActorCommand
+  case class ReadExecutionResult(hash: String) extends CacheActorCommand
+  case class StoreExecutionResult(hash: String, succeededResponse: SucceededResponse) extends CacheActorCommand
+
+  trait CacheActorResponse
+  case class ExecutionResultFound(succeededResponse: SucceededResponse) extends CacheActorResponse
+  case object ExecutionResultNotFound extends CacheActorResponse
+  case class ExecutionResultStored(hash: String) extends CacheActorResponse
+  case object ExecutionResultAlreadyExist extends CacheActorResponse
+
+}
+
+trait CacheActor extends Actor with ActorLogging {
+  def tag: String = "[CacheActor]"
+  def cacheEnabled: Boolean = false
+  def forceRewrite: Boolean = false
+
+  override def receive: Receive = {
+    case ReadExecutionResult(hash) =>
+      try {
+        if (cacheEnabled) {
+          val executionResult = readExecutionResult(hash)
+          log.info(s"{} Execution result found in cache for hash {}. Returning result: {}.", tag, hash, executionResult)
+          sender() ! ExecutionResultFound(executionResult.succeededResponse)
+        } else {
+          log.warning("{} Cache is not enabled. Returning ExecutionResultNotFound message.", tag)
+          sender() ! ExecutionResultNotFound
+        }
+      } catch {
+        case ex: CachedResultNotFoundException => sender() ! ExecutionResultNotFound
+      }
+
+    case StoreExecutionResult(hash, succeededResult) =>
+      try {
+        if (cacheEnabled) {
+          storeExecutionResult(CachedExecutionResult(hash, succeededResult))
+          log.info(s"{} Cache entry for job [{}] stored successfully.", tag, hash)
+          sender() ! ExecutionResultStored(hash)
+        } else {
+          log.warning("{} Cache is not enabled. Returning ExecutionResultAlreadyExist message.", tag)
+          sender() ! ExecutionResultAlreadyExist
```

Comment:
Is this a correct msg to send in this scenario when the flag itself is off? The name / response seems off.